tools/proofers/for_mentors.php

```diff
@@ -171,6 +174,101 @@ function output_project_details($mentored_round, $projectid, $nameofwork, $autho
     output_page_list_table($mentored_round, $projectid);
 }
 
+/**
+ * Checkout/return pages for a project.
+ *
+ * We do this per-project instead of at the top so that messages about the
+ * checkout are included with the project on the page.
+ */
+function checkout_return_project_pages(Round $mentored_round, string $projectid)
+{
+    // return if the user hasn't requested checkouts/returns for this project
+    if ((!isset($_POST["checkout_proofreader"]) && !isset($_POST["return_proofreader"]))
+        || get_projectID_param($_POST, "projectid") != $projectid) {
+        return;
+    }
+
+    // get the list of pages to checkout/return for the given user
+    $checkout_pages = [];
+    $return_pages = [];
+    if (isset($_POST["checkout_proofreader"])) {
+        // pages for proofreader
+        validate_projectID($projectid);
+        $sql = sprintf(
+            "
+            SELECT image
+            FROM $projectid
+            WHERE $mentored_round->user_column_name = '%s'
+                AND state like '%%avail'
+            ",
+            DPDatabase::escape($_POST["proofreader"])
+        );
+        $res = DPDatabase::query($sql);
+        while ([$image] = mysqli_fetch_row($res)) {
+            $checkout_pages[] = $image;
+        }
+    } elseif (isset($_POST["return_proofreader"])) {
+        // pages for proofreader
+        validate_projectID($projectid);
+        $sql = sprintf(
+            "
+            SELECT image
+            FROM $projectid
+            WHERE $mentored_round->user_column_name = '%s'
+                AND state like '%%out'
+                AND {$mentored_round->mentor_round->user_column_name} = '%s'
+            ",
+            DPDatabase::escape($_POST["proofreader"]),
+            DPDatabase::escape(User::current_username())
+        );
+        $res = DPDatabase::query($sql);
+        while ([$image] = mysqli_fetch_row($res)) {
+            $return_pages[] = $image;
+        }
+    }
+
+    // try to check out / return each one for the current user
+    // LPage enforces that the user has the ability to checkout or return the
+    // page, including that the page is in a valid state for that action
+    $project = new Project($projectid);
+    $pages_checked_out = 0;
+    $pages_returned = 0;
+    $errors = [];
+    foreach ($checkout_pages as $image) {
+        try {
+            $lpage = get_indicated_LPage($project->projectid, $project->state, $image);
```

Comment:
```suggestion
            $lpage = new LPage($project, $image, $page_state, 0);
```
This is how LPage does this for checking out a page.